src/modules/edc-demo/pages/catalog-browser/catalog-browser.component.ts

```diff
@@ -28,14 +31,31 @@ export class CatalogBrowserComponent implements OnInit {
   runningNegotiations: Map<string, NegotiationResult> = new Map<string, NegotiationResult>(); // contractOfferId, NegotiationResult
   finishedNegotiations: Map<string, ContractNegotiationDto> = new Map<string, ContractNegotiationDto>(); // contractOfferId, contractAgreementId
   private fetch$ = new BehaviorSubject<SearchParams>({label: '', location: ''});
+  public url: string = "";
+  public ownAssets: boolean = true;
 
   constructor(private apiService: CatalogBrowserService,
               public dialog: MatDialog,
-              public titleService: Title) {
+              public titleService: Title,
+              private authenticationService: AuthenticationService,
+              private route: ActivatedRoute) {
   }
 
   ngOnInit(): void {
-    this.filteredContractOffers$ = this.apiService.getContractOffers();
+    this.authenticationService.userProfile$.subscribe(userProfile => {
+      if (!userProfile) {
+        throw new Error('UserProfile is null or undefined.');
+      }
+      this.url = userProfile.url;
+      this.route.data.subscribe(data => {
+        if (data.title === 'Meine Assets') {
```

Comment:
So hoert das auf zu funktionieren wenn einer der beiden Routen mal nen anderen Titel bekommt. Besser eigenen Parameter (boolean falls aktuell und in Zukunft sicher nur zwei Werte, enum falls akutell oder potenziell in Zukunft >2 Werte) dafuer verwenden.

Reply:
Wie genau meinst du das? Die eigentliche Logik liegt ja im app-routing.module. Also wenn, dann müsste man den Parameter doch in dieses JSON-Skelett packen oder nicht?

  {
    path: 'meine-assets',
    component: CatalogBrowserComponent,
    data: {title: 'Meine Assets', icon: 'home' }
  },
  {
    path: 'catalog-browser',
    component: CatalogBrowserComponent,
    data: {title: 'Katalog', icon: 'sim_card' }
  },

Da könnte man jetzt ein drittes Key-Value-Pair hinzufügen und im catalog-browser auslesen, aber da könnte man ja genauso argumentieren, dass sich da mal was ändern könnte. Da finde ich die aktuelle Variante etwas mehr straightforward.

Reply:
Neuer Stand aus anderem Kommentar: https://github.com/agri-gaia/DataDashboard/pull/11/commits/faf3d71c696450b8571834fef0eb9f8bd7e2f545

Reply:
https://github.com/agri-gaia/DataDashboard/pull/11/commits/c66ab1ba0aace87f49ef38ad5dfe1161d9920ee3